packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +100,93 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// A controller for the content of a [Selectable].
+///
+/// A user can set the value on this controller to modify the content under
+/// a selection created by [SelectionArea] or [SelectableRegion].
+abstract class SelectedContentController<T extends Object> extends ValueNotifier<T> {
```

Comment:
Should the type parameter be an InlineSpan or can it really be anything?

Reply:
I made it generic here in case we plan to add support for a different type of format down the road. For example, if Text widgets move from `InlineSpan` to some new type of annotated string, or if someone has a Selectable state that they only track as a `String`.